### PR TITLE
Active sessions endpoint

### PR DESCRIPTION
**Pull request by gitjul**
This PR contains a proper refactor of fixed active sessions - added a new endpoint in a similar fashion to the dormant active sessions endpoint and cleaned up obsolete stuff. It also renames streaming to active and non-streaming to dormant everywhere.

### CODE REVIEW

File: app/controllers/api/realtime/sessions_controller.rb [hunk 1]
```diff
@@ -30,17 +30,6 @@ class SessionsController < BaseController
 
       respond_to :json
 
-      def index_streaming
-        data = decoded_query_data(params[:q])
-        INT_Q_ATTRS.each { |key| data[key] = data[key].to_i if data.key?(key) }
```
Comment (AkkarinForest) [inline app/controllers/api/realtime/sessions_controller.rb]: I guess you can also remove INT_Q_ATTRS on line 24

File: app/controllers/api/fixed/active/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module Api
+  module Fixed
+    module Active
+      class SessionsController < BaseController
+        respond_to :json
+        INT_Q_ATTRS = %i[time_from time_to day_from day_to limit offset]
```
Comment (AkkarinForest) [inline app/controllers/api/fixed/active/sessions_controller.rb]: is this needed?

File: app/javascript/elm/src/Main.elm [hunk 3]
```diff
@@ -183,6 +183,12 @@ init flags url key =
         , overlay = Overlay.init flags.isIndoor
         , scrollPosition = flags.scrollPosition
         , theme = Theme.toTheme flags.theme
+        , status =
+            if flags.isActive then
+                Active
+
+            else
+                Dormant
```
Comment (AkkarinForest) [inline app/javascript/elm/src/Main.elm]: You could move this to the Status module in a similar fashion like theme works
```
toStatus status =
    case status of
        True ->
            Active

        False ->
            Dormant

```

File: app/javascript/elm/src/Main.elm [hunk 4]
```diff
@@ -417,13 +423,21 @@ update msg model =
...
                 ( subModel, subCmd ) =
                     deselectSession model
+
+                newStatus =
+                    case model.status of
+                        Active ->
+                            Dormant
+
+                        Dormant ->
+                            Active
```
Comment (AkkarinForest) [inline app/javascript/elm/src/Main.elm]: Same here, you can extract this into a function in a status module
```
toggle : Theme -> Theme
toggle theme =
    case theme of
        Default ->
            Blue

        Blue ->
            Default
```

File: app/javascript/elm/src/Main.elm [hunk 5]
```diff
@@ -13,6 +13,7 @@ import Data.Page exposing (Page(..))
 import Data.Path as Path exposing (Path)
 import Data.SelectedSession as SelectedSession exposing (SelectedSession)
 import Data.Session exposing (..)
+import Data.Status exposing (Status(..))
```
Comment (AkkarinForest) [inline app/javascript/elm/src/Main.elm]: In general, you could create helpers functions in a Status module in a way that you wouldn't even have to expose the internals of status - I mean the `(..)`
It's more code, but I'm a fan of these small methods. It feels right :)
You can even write very small thingies like `default = Active`
Check next comments and a Theme module if you want to see what I mean.

File: app/models/api/active_session.rb [hunk 6]
```diff
@@ -1,7 +1,7 @@
 require 'dry-validation'
 require 'dry-struct'
 
-module Api::DormantSessions
+module Api::ActiveSessions
```
Comment (AkkarinForest) [inline app/models/api/active_session.rb]: Is this file used?

Comment (gitjul) [reply app/models/api/active_session.rb]: no:)

File: app/controllers/api/fixed/active/sessions_controller.rb (hunk 2 above)
Comment (gitjul) [reply app/controllers/api/fixed/active/sessions_controller.rb]: no:)

**Pull request merged**